src/zkevm_specs/state.py

```diff
@@ -518,6 +532,19 @@ def keys_rwc_to_limbs_in_order(row: Row) -> List[FQ]:
     if row.is_write == 0 and all_keys_eq(row, row_prev):
         assert row.value == row_prev.value
 
+    # 0.6. value_prev should be correct
+    if all_keys_eq(row, row_prev):
+        reversible_rws = [
+            Tag.Storage,
+            Tag.Account,
+            Tag.TxRefund,
+            Tag.TxAccessListAccount,
+            Tag.TxAccessListAccountStorage,
+            Tag.TxRefund,
+        ]
+        if row.tag() in reversible_rws:
+            assert row.value_prev == row_prev.value
```

Comment:
Isn't this constraint already enforced in `check_account` and `check_storage` for `Tag.Account` and `Tag.Storage`?

On a second note, writing this constraint only for specific tags is more complex than writing for all tags. Maybe we can have this constraint for all tags (even when it's not needed), and just make sure the witness we generate is valid.

Reply:
I agree completely with that!

If we can make a generic constraint that works over all of them at once that would make it much simpler.